src/cdk/v2/destinations/sprig/procWorkflow.yaml

```diff
@@ -0,0 +1,71 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+
+steps:
+  - name: messageType
+    template: |
+      .message.type.toLowerCase();
+
+  - name: validateInput
+    template: |
+      let messageType = $.outputs.messageType;
+      $.assert(messageType, "message Type is not present. Aborting");
+      $.assert(messageType in {{$.EventType.([.IDENTIFY, .TRACK])}}, "message type " + messageType + " is not supported");
+      $.assertConfig(.destination.Config.apiKey, "API Key is not present. Aborting");
+
+  - name: prepareIdentifyPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      $.context.payload = .message.({
+        userId: {{{{$.getGenericPaths("userIdOnly")}}}},
+        emailAddress: {{{{$.getGenericPaths("email")}}}},
+        attributes: .context.traits
+      });
+      $.context.payload = $.removeUndefinedAndNullValues($.context.payload);
+  
+  - name: validateIdentifyPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      $.assert($.context.payload.userId, "userId is required");
+
+  - name: prepareTrackPayload
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.context.payload = .message.({
+        userId: {{{{$.getGenericPaths("userIdOnly")}}}},
+        emailAddress: {{{{$.getGenericPaths("email")}}}}
+      });
+      const events = [
+        {
+          event: .message.event,
+          timestamp: $.toMilliseconds(.message.().({{{{$.getGenericPaths("timestamp")}}}}))
+        }
+      ]
+      $.context.payload.events = events;
+      $.context.payload = $.removeUndefinedAndNullValues($.context.payload);
+
+  - name: validateTrackPayload
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.assert($.context.payload.userId, "userId is required");
+      $.assert($.context.payload.events[0].event, "event name is required");
```

Comment:
is it better to validate .message.event instead?

Reply:
addressed